mission_control/include/mission_control/behavior.h

```diff
@@ -32,370 +32,77 @@
-        std::cout << "No attributes for when tag" << std::endl;
-      }
-    }
-    else if (xmlParamTag.compare("timeout") == 0)
+    if (status() == BT::NodeStatus::RUNNING)
     {
-      std::map<std::string, std::string> attribMap = it->getXMLTagAttribute();
-      if (attribMap.size() > 0)
-      {
-        std::map<std::string, std::string>::iterator attrib_it;
-        attrib_it = attribMap.begin();
-        pnts2 = parseNodeTimeStamp(attrib_it->second, it->getXMLTagValue(), m_timeout);
-        m_timeout_ena = true;
-      }
-      else
-      {
-        std::cout << "No attributes for timeout tag" << std::endl;
-      }
+      ROS_INFO_STREAM("HALT");
     }
-
-    if ((pnts1 == false) || (pnts2 == false)) retval = false;
-
-    return retval;
+    setStatus(BT::NodeStatus::IDLE);
   }
 
-  bool parseNodeTimeStamp(std::string unit, std::string val, TimeStamp& t)
-  {
-    using boost::gregorian::date;
-    using boost::posix_time::time_duration;
-    using boost::posix_time::ptime;
-    using boost::posix_time::from_iso_string;
-    int hrs, min, sec;
-    bool ret = true;
-
-    if (unit.compare("abs") == 0)
-    {
-      // Parse an absolute time in HH:MM:SS format
-      if (sscanf((const char*)val.c_str(), "%d:%d:%d", &hrs, &min, &sec) != 3)
-      {
-        std::cout << "Could not parse \"abs\" time." << std::endl;
-        ret = false;
-      }
-      else
-      {
-        t.type = TimeStamp::ABS_TIME;
-        t.time = ptime(date(0, 0, 0), time_duration(hrs, min, sec));
-      }
-    }
-    else if (unit.compare("hms") == 0)
-    {
-      // Parse a relative time in HH:MM:SS format
-      if (sscanf((const char*)val.c_str(), "%d:%d:%d", &hrs, &min, &sec) != 3)
-      {
-        std::cout << "Could not parse \"hms\" time." << std::endl;
-        ret = false;
-      }
-      else
-      {
-        t.type = TimeStamp::REL_TIME;
-        t.duration = time_duration(hrs, min, sec);
-      }
-    }
-    else if (unit.compare("iso") == 0)
-    {
-      // Parse an absolute ISO 8601 date/time string
-      t.type = TimeStamp::ABS_DATETIME;
-      try
-      {
-        t.time = from_iso_string(val);
-      }
-      catch (std::exception)
-      {
-        std::cout << "Could not parse \"iso\" time." << std::endl;
-        ret = false;
-      }
-    }
-    else if (unit.compare("sec") == 0)
-    {
-      // Parse a relative time in seconds
-      if (sscanf((const char*)val.c_str(), "%d", &sec) != 1)
-      {
-        std::cout << "Could not parse \"sec\" time." << std::endl;
-        ret = false;
-      }
-      else
-      {
-        t.type = TimeStamp::REL_TIME;
-        t.duration = time_duration(0, 0, sec);
-      }
-    }
-    else
-    {
-      ROS_INFO("parseNodeTimeStamp - unsupported time unit [%s]", unit.c_str());
-
-      ret = false;
-    }
-    return ret;
-  }
+  /// Method (to be implemented by the user) that sends msg to actuators.
+  /// It's used in the transition from IDLE to RUNNING
+  virtual void publishMsg() = 0;
```

Comment:
@patapufate does this method still make sense?

Reply:
After meeting. This method is particular for each "Action". Therefor, we remove from behavior object and move it to each "action" as private method.